### ISSUE TITLE
Foreach with null coalescing operator

### ISSUE THREAD

**Issue by MidnightDesign** [kalessil/phpinspectionsea#606]
```php
function foo(?\Traversable $a)
{
    foreach ($a ?? [] as $b) {
        echo $b;
    }
}
```

With the above code I get this false positive:
https://github.com/kalessil/phpinspectionsea/blob/master/docs/types-compatibility.md#foreach-source-to-iterate-over

It doesn't report the error with the type hint `?array` and when replacing the `??` operator with `?:`.

**Comment by kalessil** [comment]
That's something for reporting to PhpStorm team: `$a ?? []` type resolved as `array|null|\Traversable` (I didn;t check if this has been fixed in EAP already).

Following workaround can be applied:
```php
function foo(?\Traversable $a)
{
    $source = $a ?? [];
    foreach (source  as $b) {
        echo $b;
    }
}
```

**Comment by kalessil** [comment]
@MidnightDesign : if you decide to report the bug, please share back-link here.

**Comment by kalessil** [comment]
Popped up again in #774, we'll go with a workaround.
Also PS ticket: https://youtrack.jetbrains.com/issue/WI-37013

**Comment by kalessil** [comment]
@MidnightDesign : fixed, will be in the next release.
